contrib/dunst_xr_theme_changer.sh

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env bash
+###############################################################################
+#
+# This script creates a dunst user configuration file in
+# $XDG_CONFIG_HOME/dunst/ folder, changing dunst basic theming options
+# according to your current X resources color palette.
+#
+# It gets your current user configuration file
+# -$HOME/.conf/dunst/dunstrc, which is copied from the default
+# config file in /usr/share/dunst/dunstrc if it does not exist- and
+# changes the defined attributes -declared further on in the script as
+# $theme_attr_dict dictionary- to the new ones. Then it dumps the new
+# configuration to $user_xr_color_conf file.
+#
+###############################################################################
+
+set -e
+
+script_name="$(basename "$0")"
+
+# Function to get resource values
+xrdb_get () {
+    output="$(xrdb -q | grep -e "$1" | cut -f2)"
```

Comment:
> The generated file has the right colours, but the string is terminated on a new line, and the colour is repeated in it.

I think here is the problem. You have to make sure, that only a single line is retrieved. Currently you grep for anything matching the string `$1`. **It could be even the value, which matches.**. Usually Xresources have `*color0: #abcdef`, but you also could define `XTerm*color0: #123456`. And your regex would match both.

As @nhatzHK has already mentioned: You should use another method to query the X resource database, which also takes care of the precedence of specific values.

@nhatzHK Just to verify my thesis: What does `xrdb -q | grep 'color0:'` give as the output?

Reply:
I think this is right. I do have several lines ending with `colorX` in my X ressource database
```sh
$ xrdb -q | grep 'color0:'
*.color0:	#0d170a  ## notice the leading dot
*color0:	#0d170a
```

Reply:
The thing is: should I just get the first line?

e.g.: If you have

```bash
XTerm*color0:	#123456
*.color0:	#0d170a
*color0:	#0d170a
```

then with `head -n1` will get the XTerm color:

```bash
$ xrdb -q | grep "color0:" | head -n1 | cut -f 2
#123456
```

So maybe it is interesting to place colors at the begining or at the end. Other option is to write the exact xresource name:

```bash
$ xrdb -q | grep "*.color0:" | head -n1 | cut -f 2
#0d170a
```

But maybe this option will get the same problem that we have before, something like

```bash
XTerm*.color0:	#123456
*.color0:	#0d170a
```

What do you think?

Reply:
How about we throw more regex at it?
```
^(?:\*?\.?[a-zA-Z0-9]+):[[:space:]]?(.*)$
or
^(?:\*?\.?color0):[[:space:]]?(.*)$
```
to catch wildcards and
```
^(?:Dunst\*?\.?[a-zA-Z0-9]+):[[:space:]]?(.*)$
or
^(?:Dunst\*?\.?color0):[[:space:]]?(.*)$
```
for dunst specifically.

Reply:
Hey guys, if I read the docs right, the key should be the `appres` program. No regex involved.

Reply:
Hum... Can't figure out how to make it work :(

Reply:
This should be as simple as replacing `xrdb -q` with `appres Dunt` and then using ` | head -n1` as you suggested. Some testing shows that it always puts the more specific match at the top.

Reply:
Thanks Nikos! I was trying something like `Dunst.color0` and wasn't working. The script is fixed now.